### PR TITLE
fix: OPTIC-183: Datamanager filters not retrieving all task data results

### PR DESCRIPTION
**Pull request by bmartel**
#### What does this fix?

When a project using a cloud storage source would at some point later use a manual upload of some data, the cloud storage data would become no longer filterable in DataManager. This fixes that issue by providing support for both the placeholder `$undefined$` field as well as the named field provided from the label config for any filters for that particular field.

#### What feature flags were used to cover this change?
- `fflag_fix_back_optic_183_datamanager_filter_placeholder_keyed_task_data_short`

### CODE REVIEW

File: label_studio/data_manager/managers.py [hunk 1]
```diff
@@ -382,42 +455,67 @@ def apply_filters(queryset, filters, project, request):
...
+                        f'{alt_field_name}__gte': _filter.value.min,
+                        f'{alt_field_name}__lte': _filter.value.max,
                     }
-                ),
-            )
+                )
+            filter_expressions.append(q)
 
-        # not in
+        # not_in
```
Comment (bmartel) [inline label_studio/data_manager/managers.py]:
```suggestion
        # not in
```

File: label_studio/data_manager/managers.py [hunk 2]
```diff
@@ -243,13 +243,28 @@ def add_user_filter(enabled, key, _filter, filter_expressions):
...
     if not filters:
         return queryset
 
     # convert conjunction to orm statement
     filter_expressions = []
     custom_filter_expressions = load_func(settings.DATA_MANAGER_CUSTOM_FILTER_EXPRESSIONS)
+    preprocess_field_name = load_func(settings.PREPROCESS_FIELD_NAME)
+    preprocess_filter = load_func(settings.DATA_MANAGER_PREPROCESS_FILTER)
+    handle_alt_fieldname = flag_set(
+        'fflag_fix_back_optic_183_datamanager_filter_placeholder_keyed_task_data_short', user=request.user
```
Comment (bmartel) [inline label_studio/data_manager/managers.py]:
```suggestion
        'fflag_fix_back_optic_183_datamanager_filter_placeholder_keyed_task_data_short', user='auto'
```

**Review by wesleylima — APPROVED** [review]

**Review by dredivaris — APPROVED** [review]
LTGM and got the gist of it but tbh, those queries melted my brain

**Comment by bmartel** [comment]
> LTGM and got the gist of it but tbh, those queries melted my brain

You and me both, this is also why I generated a refactor ticket before even tackling this issue, it was already very difficult to parse.

**Comment by robot-ci-heartex** [comment]
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

**Comment by bmartel** [comment]
Closing as not needed at this time. Will reopen and re-evaluate if necessary but for now this is stale and non-actionable.

**Pull request closed without merging**